libbeat/processors/add_host_metadata/add_host_metadata.go

```diff
@@ -75,28 +67,41 @@ func newHostMetadataProcessor(cfg *common.Config) (processors.Processor, error)
 
 // Run enriches the given event with the host meta data
 func (p *addHostMetadata) Run(event *beat.Event) (*beat.Event, error) {
-	p.loadData()
+	err := p.loadData()
+	if err != nil {
+		return nil, err
+	}
+
 	event.Fields.DeepUpdate(p.data.Get().Clone())
 	return event, nil
 }
 
 func (p *addHostMetadata) expired() bool {
+	if p.config.CacheTTL <= 0 {
+		return true
+	}
+
 	p.lastUpdate.Lock()
 	defer p.lastUpdate.Unlock()
 
-	if p.lastUpdate.Add(cacheExpiration).After(time.Now()) {
+	if p.lastUpdate.Add(p.config.CacheTTL).After(time.Now()) {
 		return false
 	}
 	p.lastUpdate.Time = time.Now()
 	return true
 }
 
-func (p *addHostMetadata) loadData() {
+func (p *addHostMetadata) loadData() error {
 	if !p.expired() {
-		return
+		return nil
+	}
+
+	h, err := sysinfo.Host()
+	if err != nil {
+		return err
 	}
 
-	data := host.MapHostInfo(p.info)
```

Comment:
IIUC the problem was here b/c it never fetched updated info

Reply:
Correct